src/net/primitives.c

```diff
@@ -142,14 +141,14 @@ int8_t _z_undeclare_publisher(_z_publisher_t *pub) {
 /*------------------ Subscriber Declaration ------------------*/
 _z_subscriber_t *_z_declare_subscriber(_z_session_t *zn, _z_keyexpr_t keyexpr, _z_subinfo_t sub_info,
                                        _z_data_handler_t callback, _z_drop_handler_t dropper, void *arg) {
-    _z_subscription_t *rs = (_z_subscription_t *)z_malloc(sizeof(_z_subscription_t));
-    rs->_id = _z_get_entity_id(zn);
-    rs->_key = _z_get_expanded_key_from_key(zn, _Z_RESOURCE_IS_LOCAL, &keyexpr);
-    rs->_info = sub_info;
-    rs->_callback = callback;
-    rs->_dropper = dropper;
-    rs->_arg = arg;
-    if (_z_register_subscription(zn, _Z_RESOURCE_IS_LOCAL, rs) < 0) {
+    _z_subscription_t s;
+    s._id = _z_get_entity_id(zn);
+    s._key = _z_get_expanded_key_from_key(zn, _Z_RESOURCE_IS_LOCAL, &keyexpr);
+    s._info = sub_info;
+    s._callback = callback;
+    s._dropper = dropper;
+    s._arg = arg;
+    if (_z_register_subscription(zn, _Z_RESOURCE_IS_LOCAL, &s) < 0) {
```

Comment:
`z_subscription_t` here can be in the stack avoiding additional heap allocations.
Only when adding it into the list of subscriptions, it will be copied into the heap by means of the smart pointer creation.